### PR TITLE
fix: max withdrawal bugs

### PR DESCRIPTION
**Pull request by avniculae**

### CODE REVIEW

File: src/components/composite/MarginAmount/MarginAmount.tsx [hunk 1]
```diff
@@ -50,12 +50,16 @@ const MarginAmount: React.FunctionComponent<MarginAmountProps> = ({
...
 
-  const maxAmountToWithdraw =
+  let maxAmountToWithdraw =
     !isUndefined(marginAmount) && !isUndefined(initialMarginRequirement)
-      ? marginAmount - initialMarginRequirement
+      ? Math.max(marginAmount - initialMarginRequirement, 0)
       : undefined;
+  if (maxAmountToWithdraw) {
+    maxAmountToWithdraw = Math.floor(maxAmountToWithdraw * 10000) / 10000;
+  }
```
Comment (ssbarbee) [inline src/components/composite/MarginAmount/MarginAmount.tsx:59]: you can merge this Math.max and Math.floor in one line and make maxAmountToWithdraw a const. WDYT?

File: src/hooks/useCurrentPositionMarginRequirement.ts [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
     useState<number>();
 
   // Load the users balance of the required token
   useEffect(() => {
     const getData = async () => {
-      try {
-        const mrm = await amm.getPositionMarginRequirement(1, 999);
-        setCurrentPositionMarginRequirement(mrm);
-      } catch (e) {
+      if (fixedRateLower && fixedRateUpper) {
```
Comment (ssbarbee) [inline src/hooks/useCurrentPositionMarginRequirement.ts:15]: is the check isUndefined missing here?

Comment (avniculae) [reply src/hooks/useCurrentPositionMarginRequirement.ts:15]: doesn't it have the same effect?

Comment (ssbarbee) [reply src/hooks/useCurrentPositionMarginRequirement.ts]: currently it also protects you against 0s, nulls, empty strings, not sure if this is desired

undefined && undefined === false
0 && 0 === false
null && null === false
'' && '' === false

**Review by ssbarbee — APPROVED** [review]

**Pull request merged**